src/lib.rs

```diff
@@ -233,12 +233,10 @@ pub trait Outgoing {
 }
 
 /// Gives users the ability to define their own serializable/deserializable errors.
-pub trait EndpointError {
-    /// The specific Error type when deserialization fails.
-    type E;
+pub trait EndpointError: Sized {
     /// 
-    fn try_from_error(self, response: http::Response<Vec<u8>>)
-        -> Result<Self::E, error::ResponseDeserializationEndpointError>;
+    fn try_into_error(self, response: http::Response<Vec<u8>>)
```

Comment:
Oh, I missed this before. This should be a constructor like `TryFrom::try_from`, not a method.

```suggestion
    fn try_from_response(response: http::Response<Vec<u8>>)
```
```